src/ophyd_async/epics/adcore/_core_writer.py

```diff
@@ -152,30 +153,21 @@ async def get_indices_written(self) -> int:
     async def collect_stream_docs(
         self, name: str, indices_written: int
     ) -> AsyncIterator[StreamAsset]:
+        if self._path_info is None:
+            raise RuntimeError("Writer must be opened before collecting stream docs!")
```

Comment:
https://github.com/bluesky/ophyd-async/pull/931 added a utility that would help here

Reply:
```suggestion
        path_info = error_if_none(self._path_info, "Writer must be opened before collecting stream docs!")
```